sound/soc/sof/pm.c

```diff
@@ -233,12 +240,36 @@ static int sof_resume(struct device *dev)
 {
 	struct sof_platform_priv *priv = dev_get_drvdata(dev);
 	struct snd_sof_dev *sdev = dev_get_drvdata(&priv->pdev_pcm->dev);
+	struct hdac_bus *bus = sof_to_bus(sdev);
+	struct hdac_ext_link *hlink = NULL;
 	int ret = 0;
 
 	/* do nothing if dsp resume callback is not set */
 	if (!sdev->ops->resume)
 		return 0;
 
+	/*
+	 * clear TCSEL to clear playback on some HD Audio
```

Comment:
pm.c is a generic file so should only contain generic code. These APL PCI writes should be an intel specific file and wrapped as some PM ops. They may already be part of existing functions too, so some parts could also be reused.